src/azure-cli/azure/cli/command_modules/util/custom.py

```diff
@@ -130,19 +136,37 @@ def upgrade_version(cmd, update_all=None, yes=None):  # pylint: disable=too-many
         else:
             logger.warning(UPGRADE_MSG)
     if exit_code:
-        telemetry.set_failure("CLI upgrade failed.")
+        err_msg = "CLI upgrade failed."
+        logger.warning(err_msg)
+        telemetry.set_failure(err_msg)
         sys.exit(exit_code)
-    # Updating Azure CLI and extension together is not supported in homebrewe package.
+
+    import azure.cli.core
+    import importlib
+    importlib.reload(azure.cli.core)
+    new_version = azure.cli.core.__version__
+
+    if update_cli and new_version == local_version:
+        err_msg = "CLI upgrade failed or aborted."
```

Comment:
Why not prompt users `The local version is already the new version and does not need to be updated` here, but report an error?